test/end2end.py

```diff
@@ -261,6 +264,29 @@ def test_option_universe(self):
         # re-set pbs environment
         set_PBS_env()
 
+
+    def test_env_variables(self):
+        """ Test the passing of (extra) variables """
+        fake_mpirun_env = """#!/bin/bash
+        echo 'fake mpirun called with args:' $@
+        echo `env`
+        """
+        install_fake_mpirun('mpirun', self.tmpdir, txt=fake_mpirun_env)
+        command = ' '.join([
+            sys.executable,
+            self.mympiscript,
+            "--setmpi impirun",
+            "--variablesprefix=USER",
+            "hostname",
+        ])
+        ec, out = run_simple(command)
+
+        for key in nub(filter(os.environ.has_key, MPI.OPTS_FROM_ENV_BASE)):
+            self.assertTrue(key in out, "%s is not in out" % key)
+        regex = r'.*-envlist [^ ]USER.*'
+        self.assertTrue(regex.find(out), "Couldn't find variablesprefix USER in script env")
+
+
```

Comment:
please also check that `PYTHONPATH` is being passed (if it is defined in the environment), since this is what caused us to find the bug